src/rust/engine/process_execution/src/lib.rs

```diff
@@ -380,7 +380,9 @@ pub struct ProcessMetadata {
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct FallibleProcessResultWithPlatform {
   pub stdout: Bytes,
+  pub stdout_digest: Digest,
   pub stderr: Bytes,
+  pub stderr_digest: Digest,
```

Comment:
I don't think that we'll want to hold both the content and digest of these. From a memory perspective, it would be much better to avoid holding copies of stdout/stderr in memory (which we have apparently been doing here: this is directly the result type of the `ProcessResult` `Node`).

Reply:
This still needs to be fixed.

Reply:
Agreed, but it's going to take a lot of additional modification to the code (largely rewriting the `process_execution` tests) to be able to get rid of `stdout` and `stderr`, which is why I'd like to do it in a followup PR.